### PR TITLE
Add tests for <SearchPageLayout> component (#1887)

### LINKED ISSUE #1887: Add tests for <SearchPageLayout> component
Write unit tests for the `<SearchPageLayout>` React component to ensure expected behavior, edge case handling, and rendering logic.

## Essential Test Coverage Checklist

- [ ] **Renders successfully with minimal required props**  
- [ ] **Conditional rendering logic**  
- [ ] **Prop-based behavior** – different props affect output  
- [ ] **Event handling** – simulate user actions and verify callbacks  
- [ ] **State changes / internal logic**  
- [ ] **Default values and fallbacks**  
- [ ] **Text and content rendering**  
- [ ] **Handles edge cases and invalid inputs**  
- [ ] **Accessibility roles and labels**  
- [ ] **DOM structure / classNames / styles**

## Test Reference  
You can refer to the `AutoScrollToTop.test.tsx` file for an example of structure and best practices.  
To explore more examples, see the full component tests folder.

### PR DESCRIPTION
**Pull request by AKA-AKN**
## Proposed change
Resolves #1887

This PR adds comprehensive unit tests for the `<SearchPageLayout />` component.  
It includes tests for standard rendering, prop-based behavior, edge cases, event handling, and layout structure validation.

Accessibility roles and labels were **not tested**, as the component currently does not include any explicit ARIA roles or labels.

## Checklist

- [x] I've read and followed the [contributing guidelines](https://github.com/OWASP/Nest/blob/main/CONTRIBUTING.md).
- [x] I've run `make check-test` locally; all checks and tests passed.

### CODE REVIEW

File: frontend/__tests__/unit/components/SearchPageLayout.test.tsx [hunk 1]
```diff
@@ -0,0 +1,410 @@
...
+      // onSearch and onPageChange are intentionally omitted
+      searchPlaceholder="Search..."
+      empty="No results"
+      indexName="test"
+    />
+  )
+
+  expect(screen.getByPlaceholderText('Search...')).toBeInTheDocument()
+  expect(screen.getByText('No results')).toBeInTheDocument()
+})
```
[the remark this answers is not in the document]
Comment (kasya) [reply frontend/__tests__/unit/components/SearchPageLayout.test.tsx]: @AKA-AKN This is a very valid point. If the prop is required you should not just omit it. 
If you still want to not include it - you need pass a proper TypeScript error suppression
Screenshot 2025-08-01 at 6 09 40 PM

Please run `make test` locally and check errors that are showing up in the terminal. These need to be resolved 👌🏼

[the remark this answers is not in the document]
Comment (AKA-AKN) [reply frontend/__tests__/unit/components/SearchPageLayout.test.tsx]: Hi @kasya I have took care of these errors and checked in make test as well. I wanted to know out of the suggested changed by coderabbit is there anything else I need to implement?

**Comment by AKA-AKN** [comment]
Hey @kasya does that mean I have successfully contributed to my first open source project?
